alert.go

```diff
@@ -577,7 +590,29 @@ func (a *AlertNode) handleAlert(ad *AlertData) {
 	}
 }
 
-func (a *AlertNode) determineLevel(now time.Time, fields models.Fields, tags map[string]string) AlertLevel {
+func (a *AlertNode) determineLevel(now time.Time, fields models.Fields, tags map[string]string, currentLevel AlertLevel) AlertLevel {
+	// Evaluate reset expression of current level if it exists
+	if cse := a.levelResets[currentLevel]; cse != nil {
+		if pass, err := EvalPredicate(cse, a.lrScopePools[currentLevel], now, fields, tags); err != nil {
+			a.logger.Printf("E! error evaluating reset expression for current level %v: %s", currentLevel, err)
+		} else if !pass {
+			for l := len(a.levels) - 1; l >= 0; l-- {
+				se := a.levels[l]
+				if se == nil {
+					continue
+				}
+				if pass, err := EvalPredicate(se, a.scopePools[l], now, fields, tags); err != nil {
+					a.logger.Printf("E! error evaluating expression for level %v: %s", AlertLevel(l), err)
+					continue
+				} else if pass {
+					if currentLevel < AlertLevel(l) {
+						return AlertLevel(l)
+					}
+				}
+			}
```

Comment:
@nathanielc There's a little duplication of code here for checking the expressions of the levels. Can you recommend on how to refactor this?